tests/samplers_tests/test_nsga2.py

```diff
@@ -477,3 +478,117 @@ def test_call_after_trial_of_random_sampler() -> None:
     ) as mock_object:
         study.optimize(lambda _: 1.0, n_trials=1)
         assert mock_object.call_count == 1
+
+
+def test_crossover() -> None:
+    NSGAIISampler()
+    NSGAIISampler(crossover="uniform")
+    NSGAIISampler(crossover="blxalpha")
+    NSGAIISampler(crossover="sbx")
+    NSGAIISampler(crossover="vsbx")
+    NSGAIISampler(crossover="undx")
+    NSGAIISampler(crossover="undxm")
+    NSGAIISampler(crossover="spx")
+
+    with pytest.raises(ValueError):
+        NSGAIISampler(crossover="no_imp_crossover")
+
+
+parametrize_nsga2_sampler = pytest.mark.parametrize(
+    "sampler_class",
+    [
+        NSGAIISampler(population_size=2, crossover="uniform"),
+        NSGAIISampler(population_size=2, crossover="blxalpha"),
+        NSGAIISampler(population_size=2, crossover="sbx"),
+        NSGAIISampler(population_size=2, crossover="vsbx"),
+        NSGAIISampler(population_size=3, crossover="undx"),
+        NSGAIISampler(population_size=3, crossover="spx"),
```

Comment:
Yes, `undxm` is intentionally left out.
This is because, unlike other crossover operators, `undxm` will cause errors depending on the number of dimensions of the search space. (In `undxm`, the number of parent individuals used for crossover is `m`, and the number of dimensions of the search space is `n`, where `n > m-2`).
Therefore, to test for `ValueError`, we remove `undxm` around line 500 and test it separately with `test_undxm_objective` and `test_undxm_dim` functions.

But
- Not mentioned in the document.
- Not enough testing.

So, we will address the above two points.

Reply:
Sounds good. Also, I think you're asserting for the condition in the implementation but maybe it's better replaced with an error (e.g. `RuntimeError`) since it's not an implementation detail but something that should be clearly communicated to the library user.